### ISSUE TITLE
Submit button being disabled

### ISSUE THREAD

**Issue by iamthestreets** [formvalidation/formvalidation#1171]
I have a form with 4 fields:
Email, Current Password, New Password, and Confirm Password

All fields are optional, but to update their password they need to enter their current password first. I do a check in the database to match passwords (using PHP) and the issue i'm running into is when i submit the form it disables the submit button and does not submit the form. If I remove the BootstrapValidator code it works fine. Any idea why it is preventing my form from being submitted?

Not sure what info you need from me, but please let me know.

**Comment by cdomkokao** [comment]
I am getting a similar thing.  I have all optional fields but if they enter a value it needs to be a valid value.  When I use the submit button it validates green because I haven't entered anything in the field but the submit button is grayed out.  If I actually enter values in the fields that are optional it then enables the button.

**Comment by nghuuphuoc** [comment]
It's fixed in the [latest code](https://github.com/formvalidation/formvalidation/archive/master.zip).
You can download the latest code and try again.

__IMPORTANT__: You might have to look at [the upgrading guide](http://formvalidation.io/news/upgrading-to-v0.6.0/) if using the latest code.

**Comment by cdomkokao** [comment]
Hmm, ok.  I will give it a shot.  I was worried that might be the issue.  I will post back if that works.

**Comment by cdomkokao** [comment]
It isn't working, I am getting Uncaught TypeError on line 2134 of the main JS file:

    data = new FormValidation.Framework.Bootstrap(this, options);

I tried slimming down my call of the validation to just be the following:

    $('.mainForm').formValidation();

And that didn't help.

Any thoughts?

**Comment by cdomkokao** [comment]
Fixed my Uncaught TypeError problem but I still can't submit the form unless I click the submit button and then add values into the fields.  If I just submit, it highlights all of the fields with green borders and the submit button is disabled but the form doesn't submit.  I have to enter a value into one of the fields.  This isn't going to work because the fields aren't required so a user doesn't have to fill them in.

What's worse is that I can fill all of the fields in and submit and the same thing happens, the first time the fields are just highlighted like they were successful but the form doesn't submit.

**Comment by nghuuphuoc** [comment]
@cdomkokao 
Can you please give me a [jsfiddle](http://jsfiddle.net/)?

**Comment by iamthestreets** [comment]
Hi,

How would you create a jsfiddle for this to mimic a form not submitting? I am still having this issue. When I fill in the fields and it highlights the fields like they were successful but the form doesn't submit and the submit button becomes disabled. If I click into one of the fields and then click out of the field the button becomes enabled and if I click submit again the form will submit.

**Comment by iamthestreets** [comment]
I found another issue that fixed my issue - https://github.com/formvalidation/formvalidation/issues/1352
but if i click enter while still in the field it does not submit the form. it is only once I have clicked out or tabbed out of the field will it submit the form.

**Comment by nghuuphuoc** [comment]
@iamthestreets 
* What is the version of plugin you're using? 0.5.3, 0.6.0 or the latest code?
* Please give me your HTML form + JS code you're using.

**Comment by iamthestreets** [comment]
I am using the latest code I believe - v0.6.1-dev, built on 2015-02-17 8:51:23 PM.

Here is my HTML form:
```html
<form name="login" id="loginForm" action="<?php $_SERVER['PHP_SELF'] ?>" method="post">
				<fieldset>
					<h2>Please Sign In</h2>
					<hr class="colorgraph">
					<div class="row">
        				<div class="col-xs-6 col-sm-6 col-md-6">
          					<a href="#" class="btn btn-lg btn-social-facebook btn-block"><i class="fa fa-facebook"></i> &nbsp;Sign in with Facebook</a>
        				</div>
        				<div class="col-xs-6 col-sm-6 col-md-6">
          					<a href="#" class="btn btn-lg btn-social-twitter btn-block"><i class="fa fa-twitter"></i> &nbsp;Sign in with Twiter</a>
        				</div>
      				</div>
      				<div class="login-or">
        				<hr class="hr-or">
        				<span class="span-or">or</span>
      				</div>
					<div class="form-group " >						
	                    <input type="username" name="username" id="username" class="form-control input-lg" placeholder="Username or Email Address" value="">
	                    <span class="help-block" id="usernameMessage"></span>                   
					</div>
					<div class="form-group">
	                    <input type="password" name="password" id="password" class="form-control input-lg" placeholder="Password" value="">
	                    <span class="help-block" id="passwordMessage"></span>                  
					</div>
					<div class="row">
						<span class="button-checkbox">
        				<div class="col-xs-6 col-sm-6 col-md-6">
          					<label class="checkbox" for="checkbox1">
			            	<input type="checkbox" id="checkbox1" name="remember" data-toggle="checkbox" class="custom-checkbox">
			            	Remember Username
          					</label>
        				</div>
        				<div class="col-xs-6 col-sm-6 col-md-6">
          					<a href="forgot-password.php" class="btn btn-link pull-right">Forgot Password?</a>
        				</div>
        				</span>
      				</div>
					<hr class="colorgraph">
					<div class="row" style="margin-bottom: 15px">
						<div class="col-xs-6 col-sm-6 col-md-6">							
	                        <input type="submit" id="submitBtn" class="btn btn-lg btn-primary btn-block" value="Sign In">
						</div>
						<div class="col-xs-6 col-sm-6 col-md-6">
							<a href="register.php" class="btn btn-lg btn-info btn-block">Register</a>
						</div>
					</div>
					<?php 
					if ($emailActivation == "true") { 
					?>
					<div class="row">
						<div class="col-md-12">
							<a href="resend_activation.php" class="btn btn-link pull-right" role="button" value="Activate">Resend activation email</a>
			  			</div>
					</div>
					<?php 
					} 
					?>
				</fieldset>
			</form>
```
Here is the JS code:
```javascript
$(document).ready(function() {	
    $('#loginForm').formValidation({
    	framework: 'bootstrap',
        message: 'This value is not valid',
        icon: {
            valid: 'glyphicon glyphicon-ok',
            invalid: 'glyphicon glyphicon-remove',
            validating: 'glyphicon glyphicon-refresh'
        },
        button: {
        	selector: '#submitBtn'
    	},
        fields: {
            username: {
            	trigger: 'blur',
            	err: '#usernameMessage',
                validators: {
                	notEmpty: {
                        message: 'The username is required and cannot be empty'
                    }
                }
            },
            password: {
            	trigger: 'blur',
            	err: '#passwordMessage',
                validators: {
                    notEmpty: {
                        message: 'The password is required and cannot be empty'
                    },
                    stringLength: {
						min: 6,
						max: 30,
					}
                }
            }
        }
    });
    
});
```

**Comment by nghuuphuoc** [comment]
> If I click into one of the fields and then click out of the field the button becomes enabled and if I click submit again the form will submit

The issue is caused by using ```trigger: 'blur'``` option.
Also, you don't need to use 
* ```button``` option because the default ```[type="submit"]``` value covers your submit button 
* ```err``` option if you don't have special reason. In that case, the error message are shown properly as not using the ```err``` option.

Try the fiddle: http://jsfiddle.net/nghuuphuoc/rn1hgvLd/2/ and let me know if you need something else.

> when did you start charging for a commercial license

You don't need to pay for a license if using v0.5.2 and earlier versions.
It's commercial from v0.5.3.

**Comment by iamthestreets** [comment]
Thanks!

I removed: 
* `button` option 
* `err` option

Is there a way that I can still use `trigger: 'blur'`? I don't really like that it gives an error immediately after I start typing. I tried adding it back to the jsfiddle and it displayed this error: `{"error": "Please use POST request"}`

**Comment by nghuuphuoc** [comment]
@iamthestreets 
>  {"error": "Please use POST request"} 
It is just an error of jsfidlle that occurs when the form is submitted to jsfidlle (because our form target, ```form action="..."```, is missing). We don't need to worry about it.

You can keep using ```trigger: 'blur'```, and trigger the event to [enable the button](http://formvalidation.io/examples/enabling-submit-button/).

Here is updated fiddle: http://jsfiddle.net/rn1hgvLd/3/
